catalog/dags/common/ingestion_server.py

```diff
@@ -65,6 +69,21 @@ def response_check_wait_for_completion(response: Response) -> bool:
     return True
 
 
+def response_check_index_readiness_check(response: Response) -> bool:
+    """
+    Handle the response for `index_readiness_check` Sensor, to await a
+    healthy Elasticsearch cluster.
+    """
+    data = response.json()
+    status = data.get("status", None)
+
+    if Variable.get("ENVIRONMENT", default_var="dev") == "prod":
+        return status == "green"
+
+    # If we are not in production, we may proceed with a yellow cluster status.
+    return status in ["green", "yellow"]
```

Comment:
This confused me for a moment because the staging ES cluster should also always be green, but then I remembered that the catalog's environment is always either `prod` or `dev` ("local", unfortunately named considering our infrastructure configuration), and it would still be `prod` even if this DAG was pointed at the staging cluster.

I don't have a suggestion for how to clarify this, it seems like just a funky bit of domain knowledge for our Airflow instance.